include/seqan3/std/charconv

```diff
@@ -0,0 +1,1010 @@
+                else
+                    buffer = append4(buffer, v);
+            }
+        }
+        else
+        {
+            // value = bbbbcccc
+            const uint32_t b = v / 10000;
+            const uint32_t c = v % 10000;
+
+            if (v < 1000000)
+            {
+                if (v < 100000)
+                    buffer = append1(buffer, b);
+                else
+                    buffer = append2(buffer, b);
+            }
+            else
+            {
+                if (v < 10000000)
+                    buffer = append3(buffer, b);
+                else
+                    buffer = append4(buffer, b);
+            }
+
+            buffer = append4(buffer, c);
+        }
+    }
+    else if (value < 10000000000000000)
+    {
+        const uint32_t v0 = static_cast<uint32_t>(value / 100000000);
+        const uint32_t v1 = static_cast<uint32_t>(value % 100000000);
+
+        const uint32_t b0 = v0 / 10000;
+        const uint32_t c0 = v0 % 10000;
+
+        if (v0 < 1000000)
+        {
+            if (v0 < 100000)
+                buffer = append1(buffer, b0);
+            else
+                buffer = append2(buffer, b0);
+        }
+        else
+        {
+            if (v0 < 10000000)
+                buffer = append3(buffer, b0);
+            else
+                buffer = append4(buffer, b0);
+        }
+
+        buffer = append4(buffer, c0);
+        buffer = append4(buffer, v1 / 10000);
+        buffer = append4(buffer, v1 % 10000);
+    }
+    else
+    {
+        const uint32_t a =
+            static_cast<uint32_t>(value / 10000000000000000);  // 1 to 1844
+        value %= 10000000000000000;
+
+        if (a < 100)
+        {
+            if (a < 10)
+                buffer = append1(buffer, a);
+            else
+                buffer = append2(buffer, a);
+        }
+        else
+        {
+            if (a < 1000)
+                buffer = append3(buffer, a);
+            else
+                buffer = append4(buffer, a);
+        }
+
+        const uint32_t v0 = static_cast<uint32_t>(value / 100000000);
+        const uint32_t v1 = static_cast<uint32_t>(value % 100000000);
+        buffer = append4(buffer, v0 / 10000);
+        buffer = append4(buffer, v0 % 10000);
+        buffer = append4(buffer, v1 / 10000);
+        buffer = append4(buffer, v1 % 10000);
+    }
+
+    return buffer;
+}
+
+template <typename _Tp, typename = void>
+struct traits_base
+{
+    using type = uint64_t;
+
+    static int width(_Tp v)
+    {
+        auto t = (64 - __builtin_clzll(v | 1)) * 1233 >> 12;
+        return t - (v < pow10_64[t]) + 1;
+    }
+
+    static char* convert(_Tp v, char* p)
+    {
+        return u64toa(v, p);
+    }
+
+    static auto& pow() { return pow10_64; }
+};
+
+template <typename _Tp>
+struct traits_base<_Tp, decltype(void(uint32_t{declval<_Tp>()}))>
+{
+    using type = uint32_t;
+
+    static int width(_Tp v)
+    {
+        auto t = (32 - __builtin_clz(v | 1)) * 1233 >> 12;
```

Comment:
okay this only works on gcc and clang.

Reply:
Do we plan to add support for anything else sometime soon?

Reply:
It's ok, because MSVC has the actual functions already implemented and they will be picked up.